docs/admin/out-of-resource.md

```diff
@@ -0,0 +1,220 @@
+condition back to `false`.
+
+### Eviction of Pods
+
+If an eviction threshold has been met, the `kubelet` will initiate the
+process of evicting pods until it has observed the signal has gone below
+its defined threshold.
+
+The `kubelet` ranks pods for eviction that have the lowest quality of
+service that are the largest consumers of the starved compute resource
+relative to the pods scheduling request.
+
+* `BestEffort` pods that consume the most of the starved resource are failed
+first.
+* `Burstable` pods that consume the greatest amount of the starved resource
+relative to their request for that resource are killed first.  If no pod
+has exceeded its request, the strategy targets the largest consumer of the
+starved resource.
+* `Guaranteed` pods that consume the greatest amount of the starved resource
+relative to their request are killed first.  If no pod has exceeded its request,
+the strategy targets the largest consumer of the starved resource.
+
+A `Guaranteed` pod is guaranteed to never be evicted because of another pod's
+resource consumption.  If a system daemon (i.e. `kubelet`, `docker`, `journald`, etc.)
+is consuming more resources than were reserved via `system-reserved` or `kube-reserved` allocations,
+and the node only has `Guaranteed` pod(s) remaining, then the node must choose to evict a
+`Guaranteed` pod in order to preserve node stability, and to limit the impact
+of the unexpected consumption to other `Guaranteed` pod(s).
+
+### Scheduler
+
+The node will report a condition when a compute resource is under pressure.  The
+scheduler views that condition as a signal to dissuade placing additional
+pods on the node.
+
+| NodeCondition    | Scheduler Behavior                               |
+| ---------------- | ------------------------------------------------ |
+| MemoryPressure | BestEffort pods are not scheduled to the node. |
+
+## Node OOM Behavior
+
+If the node experiences a system OOM (out of memory) event prior to the `kubelet` is able to reclaim memory,
+the node depends on the [oom_killer](https://lwn.net/Articles/391222/) to respond.
+
+The `kubelet` sets a `oom_score_adj` value for each container based on the quality of service for the pod.
+
+| Quality of Service | oom_score_adj |
+| -----------------  | ------------- |
+| Guaranteed | -998 |
+| BestEffort | 1000 |
+| Burstable | min(max(2, 1000 - (1000 * memoryRequestBytes) / machineMemoryCapacityBytes), 999) |
+
+If the kubelet is unable to reclaim memory prior to a node experiencing system OOM, the `oom_killer` will calculate
+an `oom_score` based on the percentage of memory its using on the node, and then add the `oom_score_adj` to get an
+effective `oom_score` for the container, and then kills the container with the highest score.
+
+The effective behavior should be that containers with the lowest quality of service that
+are consuming the largest amount of memory relative to the scheduling request should be killed first in order
+to reclaim memory.
```

Comment:
This is not always `true` in reality. We can say that the `intended` behavior is what is mentioned above, but based on the state of the system, it is possible that the kernel might choose a Burstable pod over a Besteffort pod. I have observed this a few times.